server/routes/bookings/cancel/cancelController.ts

```diff
@@ -62,22 +60,18 @@ export default class CancelVisitController {
         return res.redirect(`${paths.BOOKINGS.VISIT}/${visitDisplayId}`)
       }
 
+      const { booker, bookings } = req.session
+      const { visits } = bookings
       const visit = visits.find(v => v.visitDisplayId === visitDisplayId)
 
       await this.visitService.cancelVisit({
         applicationReference: visit.reference,
         actionedBy: booker.reference,
       })
 
-      return res.redirect(paths.BOOKINGS.CANCEL_CONFIRMATION)
-    }
-  }
+      req.session.bookingCancelled = { hasPhoneNumber: !!visit.visitContact.telephone }
```

Comment:
Main part of the change: set session data when cancelling so we know if a phone number is available